### ISSUE TITLE
Check 'allowed to show UI' with HideValidationDialog

### ISSUE THREAD

**Issue by SirBETE** [Microsoft/ALAppExtensions#204]
Every function that generates UI should check against HideValidationDialog.

ex.
TAB37 UpdateSalesLines

```
[External] UpdateSalesLines(ChangedFieldName : Text[100];AskQuestion : Boolean)
IF NOT SalesLinesExist THEN
  EXIT;

NotRunningOnSaaS := TRUE;
CASE ChangedFieldName OF
  FIELDCAPTION("Shipping Agent Code"),
  FIELDCAPTION("Shipping Agent Service Code"):
    NotRunningOnSaaS := NOT PermissionManager.SoftwareAsAService;
END;

CheckAskQuestion(ChangedFieldName, AskQuestion); /////TFS//////

IF AskQuestion THEN BEGIN
  Question := STRSUBSTNO(
      Text031 +
      Text032,ChangedFieldName);
  IF GUIALLOWED THEN
    IF NotRunningOnSaaS THEN
      IF DIALOG.CONFIRM(Question,TRUE) THEN
...
```
We don't want to see the dialog, but we can't change AskQuestion in the functioncall.
An event with Rec, ChangedFieldName, VAR AskQuestion and CurrFieldNo as parameters is also possible to avoid this dialog.

It would be better however when every function that generates UI checks against HideValidationDialog.
```
...
  Question := STRSUBSTNO(
      Text031 +
      Text032,ChangedFieldName);
IF AskQuestion THEN BEGIN
  IF GUIALLOWED THEN
    IF NotRunningOnSaaS THEN
      if not HideValidationDialog then //////MS/////
        IF DIALOG.CONFIRM(Question,TRUE) THEN
...
```

If it doens't exist yet, a `GetHideValidationDialog` and `SetHideValidationDialog` should be provided too.

**Comment by JesperSchulz** [comment]
While a very valid request, this one is too large and generic to be addressed in this repo as an issue. If you are blocked on a specific scenario, please open an issue for that scenario to be solved in this repo. For a generic solution, we'd kindly ask you to post your idea on http://aka.ms/bcideas. Do you want us to address the specific example you mention above?

**Comment by SirBETE** [comment]
Thank you for your answer.
Although I understand the complexity of this, It would be nice to have this functionality for every object that shows UI.

We shall post this on BCIdeas.
This specific example may be handled, thanks.

**Comment by JesperSchulz** [comment]
Great! You got it!
